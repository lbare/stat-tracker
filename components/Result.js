import { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity } from 'react-native';

const Result = ({ value, setValue }) => {
  return (
    <View className='flex-1 w-full justify-center'>
      <View className='h-1/4'>
        <Text className='text-7xl pb-20 font-black text-center text-blue-500 pt-1'>
          Result
        </Text>
      </View>
      <View className='h-1/4'>
        <TouchableOpacity
          className='bg-gray-100 border border-gray-500 mx-10 p-4 rounded-lg'
          onPress={() => setValue('single')}
        >
          <Text className='text-xl text-center text-black'>Single</Text>
        </TouchableOpacity>
        <TouchableOpacity
          className='bg-gray-100 border border-gray-500 mx-10 mt-5 p-4 rounded-lg'
          onPress={() => setValue('double')}
        >
          <Text className='text-xl text-center text-black'>Double</Text>
        </TouchableOpacity>
        <TouchableOpacity
          className='bg-gray-100 border border-gray-500 mx-10 mt-5 p-4 rounded-lg'
          onPress={() => setValue('triple')}
        >
          <Text className='text-xl text-center text-black'>Triple</Text>
        </TouchableOpacity>
        <TouchableOpacity
          className='bg-gray-100 border border-gray-500 mx-10 mt-5 p-4 rounded-lg'
          onPress={() => setValue('home run')}
        >
          <Text className='text-xl text-center text-black'>Home Run</Text>
        </TouchableOpacity>
        <TouchableOpacity
          className='bg-gray-100 border border-gray-500 mx-10 mt-5 p-4 rounded-lg'
          onPress={() => setValue('out')}
        >
          <Text className='text-xl text-center text-black'>Out</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Result;
