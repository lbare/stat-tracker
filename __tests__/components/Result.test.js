import React from "react";
import renderer from "react-test-renderer";

import Result from "../../components/Result";

describe("<Result />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Result value="" setValue={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
