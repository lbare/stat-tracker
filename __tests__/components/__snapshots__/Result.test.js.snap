// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "width": "100%",
      },
      {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "height": "25%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 72,
            "lineHeight": 72,
          },
          {
            "paddingBottom": 80,
          },
          {
            "fontWeight": "900",
          },
          {
            "textAlign": "center",
          },
          {
            "color": "#3b82f6",
          },
          {
            "paddingTop": 4,
          },
        ]
      }
    >
      Result
    </Text>
  </View>
  <View
    style={
      [
        {
          "height": "25%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f3f4f6",
          "borderBottomColor": "#6b7280",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#6b7280",
          "borderLeftWidth": 1,
          "borderRightColor": "#6b7280",
          "borderRightWidth": 1,
          "borderTopColor": "#6b7280",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "marginLeft": 40,
          "marginRight": 40,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Single
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f3f4f6",
          "borderBottomColor": "#6b7280",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#6b7280",
          "borderLeftWidth": 1,
          "borderRightColor": "#6b7280",
          "borderRightWidth": 1,
          "borderTopColor": "#6b7280",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Double
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f3f4f6",
          "borderBottomColor": "#6b7280",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#6b7280",
          "borderLeftWidth": 1,
          "borderRightColor": "#6b7280",
          "borderRightWidth": 1,
          "borderTopColor": "#6b7280",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Triple
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f3f4f6",
          "borderBottomColor": "#6b7280",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#6b7280",
          "borderLeftWidth": 1,
          "borderRightColor": "#6b7280",
          "borderRightWidth": 1,
          "borderTopColor": "#6b7280",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Home Run
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f3f4f6",
          "borderBottomColor": "#6b7280",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#6b7280",
          "borderLeftWidth": 1,
          "borderRightColor": "#6b7280",
          "borderRightWidth": 1,
          "borderTopColor": "#6b7280",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Out
      </Text>
    </View>
  </View>
</View>
`;
